include/seqan3/core/platform.hpp

```diff
@@ -259,8 +259,13 @@ static_assert(sdsl::sdsl_version_major == 3, "Only version 3 of the SDSL is supp
 #    endif
 #endif
 
-//!\brief Our char literals returning std::vector should be constexpr if constexpr std::vector is supported.
-#if defined(__cpp_lib_constexpr_vector) && __cpp_lib_constexpr_vector >= 201907L
+/*!\brief Our char literals returning std::vector should be constexpr if constexpr std::vector is supported.
+ *
+ * The _GLIBCXX_DEBUG statement is a workaround for a libstdc++ bug
+ * \see https://gcc.gnu.org/bugzilla/show_bug.cgi?id=104748
+ * \see https://github.com/seqan/seqan3/issues/3221
+ **/
+#if defined(__cpp_lib_constexpr_vector) && __cpp_lib_constexpr_vector >= 201907L && !defined(_GLIBCXX_DEBUG)
```

Comment:
```suggestion
#if __cpp_lib_constexpr_vector >= 201907L && (defined(_LIBCPP_VERSION) || !defined(_GLIBCXX_DEBUG))
```

It shouldn't make a difference, but I think it's more correct to use `constexpr` when `libc++` is used.
Usually, you shouldn't compile with `libc++` and `_GLIBCXX_DEBUG`, because it has no effect.

I removed the `defined(__cpp_lib_constexpr_vector)` such that the line length isn't exceeded. This check is only required if we do a `<` or `<=` comparison afterwards. A undefined value would always return true. `>`/`>=` will just return false if the value is not defined.